modules/sagemaker/sagemaker-studio/helper_constructs/sm_roles.py

```diff
@@ -0,0 +1,335 @@
+                        "s3:PutObjectAcl",
+                        "s3:GetBucketAcl",
+                        "s3:GetBucketLocation",
+                    ],
+                    resources=[
+                        "arn:aws:s3:::{}*/*".format(s3_bucket_prefix),
+                        "arn:aws:s3:::{}*".format(s3_bucket_prefix),
+                        "arn:aws:s3:::cdk*/*",
+                        "arn:aws:s3:::cdk*",
+                        "arn:aws:s3:::sagemaker*",
+                        "arn:aws:s3:::sagemaker*/*",
+                    ],
+                ),
+                iam.PolicyStatement(
+                    effect=iam.Effect.ALLOW,
+                    actions=["s3:ListBucket"],
+                    resources=[
+                        "arn:aws:s3:::{}*".format(s3_bucket_prefix),
+                        "arn:aws:s3:::cdk*",
+                        "arn:aws:s3:::sagemaker*",
+                    ],
+                ),
+                iam.PolicyStatement(
+                    effect=iam.Effect.DENY,
+                    actions=["s3:DeleteBucket*"],
+                    resources=["*"],
+                ),
+            ],
+        )
+
+        # create role for each persona
+
+        # role for Data Scientist persona
+        self.data_scientist_role = iam.Role(
+            self,
+            "data-scientist-role",
+            assumed_by=iam.CompositePrincipal(
+                iam.ServicePrincipal("lambda.amazonaws.com"),
+                iam.ServicePrincipal("sagemaker.amazonaws.com"),
+            ),
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "AmazonSSMReadOnlyAccess",
+                ),
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "AWSLambda_ReadOnlyAccess",
+                ),
+                iam.ManagedPolicy.from_aws_managed_policy_name("AWSCodeCommitReadOnly"),
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "AmazonEC2ContainerRegistryReadOnly",
+                ),
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "AmazonSageMakerFullAccess",
+                ),
+            ],
+        )
+
+        sm_deny_policy.attach_to_role(self.data_scientist_role)
+        services_policy.attach_to_role(self.data_scientist_role)
+        kms_policy.attach_to_role(self.data_scientist_role)
+        s3_policy.attach_to_role(self.data_scientist_role)
+
+        ds_role_ssm_name = "/mlops/role/ds"
+        ssm.StringParameter(
+            self,
+            "ssm-sg-ds-role",
+            parameter_name=ds_role_ssm_name,
+            string_value=self.data_scientist_role.role_arn,
+        )
+        CfnOutput(
+            self,
+            "DataScientistRoleSSMName",
+            value=ds_role_ssm_name,
+        )
+
+        # role for Lead Data Scientist persona
+        self.lead_data_scientist_role = iam.Role(
```

Comment:
Makes the code difficult to read....can you locate the outputs at the bottom together?

Reply:
Can allow